### PR TITLE
Adds a base admission acceptance test.

### PR DESCRIPTION
**Pull request by tlm**
### Checklist

 - [x] Checked if it requires a [pylibjuju](https://github.com/juju/python-libjuju) change?
 - [x] Added [integration tests](https://github.com/juju/juju/tree/develop/tests) for the PR?
 - [x] Added or updated [doc.go](https://discourse.jujucharms.com/t/readme-in-packages/451) related to packages changed?
 - [x] Do comments answer the question of why design decisions were made?

----

## Description of change

Test checks that creating external resources outside of juju are tagged
correctly with the app name.

## QA steps

*Please replace with how we can verify that the change works?*

```sh
cd tests && ./main.sh caasadmission
```

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -5,6 +5,7 @@ cmd/charmd/charmd
 cmd/charmload/charmload
 provider/ec2/internal/ec2instancetypes/index.json
 provider/ec2/internal/ec2instancetypes/data.yaml
+tests/test-*
```
Comment (SimonRichardson) [inline .gitignore:8]: :clap:

File: tests/suites/caasadmission/admission.sh [hunk 2]
```diff
@@ -0,0 +1,76 @@
+
+run_deploy_microk8s() {
+  echo
+
+  name=$1
```
Comment (SimonRichardson) [inline tests/suites/caasadmission/admission.sh:5]: Just to keep things standardized

```suggestion
  name=${1}
```

File: tests/suites/caasadmission/admission.sh [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+  export BOOTSTRAP_PROVIDER=microk8s
+  bootstrap microk8s ${name}
+
+  microk8s.config > ${TEST_DIR}/kube.conf
+}
+
+test_deploy_admission() {
+  name=test-$(petname)
+
+  run_deploy_microk8s $name
```
Comment (SimonRichardson) [inline tests/suites/caasadmission/admission.sh:15]: We need to quote everything...

```suggestion
  run_deploy_microk8s "${name}"
```

File: tests/suites/caasadmission/admission.sh [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+
+  namespace=controller-${BOOTSTRAPPED_JUJU_CTRL_NAME}
```
Comment (SimonRichardson) [inline tests/suites/caasadmission/admission.sh:17]:
```suggestion
  namespace=controller-"${BOOTSTRAPPED_JUJU_CTRL_NAME}"
```

**Comment by tlm** [comment]
$$merge$$

**Pull request merged**